packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverButton/useTeachingPopoverButtonStyles.styles.ts

```diff
@@ -1,35 +1,95 @@
+    borderBlockColor: tokens.colorNeutralForegroundOnBrand,
+    borderInlineColor: tokens.colorNeutralForegroundOnBrand,
+  },
+  brandPrimaryCarousel: {
+    color: tokens.colorBrandForeground1,
+    backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    borderBlockColor: tokens.colorTransparentBackground,
+    borderInlineColor: tokens.colorTransparentBackground,
+    ':hover': {
+      color: tokens.colorCompoundBrandForeground1Hover,
+      backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    },
+    ':hover:active': {
+      color: tokens.colorCompoundBrandForeground1Pressed,
+      backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    },
+  },
+  brandSecondary: {
+    color: tokens.colorBrandForeground1,
+    backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    ':hover': {
+      color: tokens.colorCompoundBrandForeground1Hover,
+      backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    },
+    ':hover:active': {
+      color: tokens.colorCompoundBrandForeground1Pressed,
+      backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    },
+  },
+  brandSecondaryCarousel: {
+    // In brand, this is always 'NeutralForegroundOnBrand'
+    color: tokens.colorNeutralForegroundOnBrand,
+    backgroundColor: tokens.colorBrandBackground,
+    borderBlockColor: tokens.colorNeutralForegroundOnBrand,
+    borderInlineColor: tokens.colorNeutralForegroundOnBrand,
+    ':hover': {
+      color: tokens.colorNeutralForegroundOnBrand,
+      borderBlockColor: tokens.colorNeutralForegroundOnBrand,
+      borderInlineColor: tokens.colorNeutralForegroundOnBrand,
+      backgroundColor: tokens.colorBrandBackgroundHover,
+    },
+    ':hover:active': {
+      color: tokens.colorNeutralForegroundOnBrand,
+      borderBlockColor: tokens.colorNeutralForegroundOnBrand,
+      borderInlineColor: tokens.colorNeutralForegroundOnBrand,
+      backgroundColor: tokens.colorBrandBackgroundPressed,
+    },
   },
-
-  // TODO add additional classes for different states and/or slots
 });
 
-/**
- * Apply styling to the TeachingPopoverButton slots based on the state
- */
 export const useTeachingPopoverButtonStyles_unstable = (
   state: TeachingPopoverButtonState,
 ): TeachingPopoverButtonState => {
+  const appearance = usePopoverContext_unstable(context => context.appearance);
+  const totalPages = useTeachingPopoverContext_unstable(context => context.totalPages);
+  const isCarousel = totalPages > 1;
   const styles = useStyles();
-  state.root.className = mergeClasses(teachingPopoverButtonClassNames.root, styles.root, state.root.className);
-
-  // TODO Add class names to slots, for example:
-  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
+  state.root.className = mergeClasses(teachingPopoverButtonClassNames.root, state.root.className);
+  if (state.icon) {
+    state.icon.className = mergeClasses(teachingPopoverButtonClassNames.icon, state.icon.className);
+  }
 
+  useButtonStyles_unstable(state);
```

Comment:
This may not matter as I think `TeachingPopoverButton` effectively extends `Button` but calling `useButtonStyles_unstable()` adds the `fui-Button` className to this control.

Reply:
Need this one here as the hooks are all extended separately, need to ensure 'useTeachingPopoverButtonStyles' will always implement 'useButtonStyles'